File: library/src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultCachedAudio.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class InlineQueryResultCachedAudio extends InlineQueryResult<InlineQueryResultCachedAudio> implements Serializable {
    private final static long serialVersionUID = 0L;

    private String audio_file_id;

    private String caption;
    private String parse_mode;

    public InlineQueryResultCachedAudio(String id, String fileId) {
        super("audio", id);
        this.audio_file_id = fileId;
    }

    public InlineQueryResultCachedAudio caption(String caption) {
        this.caption = caption;
        return this;
    }

    public InlineQueryResultCachedAudio parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode.name();
        return this;
    }
}

File: core/src/commonMain/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultCachedAudio.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class InlineQueryResultCachedAudio(id: String, private val audio_file_id: String) : InlineQueryResult<InlineQueryResultCachedAudio>("audio", id), Serializable {

    private var caption: String? = null
    private var parse_mode: String? = null

    fun caption(caption: String): InlineQueryResultCachedAudio {
        this.caption = caption
        return this
    }

    fun parseMode(parseMode: ParseMode): InlineQueryResultCachedAudio {
        this.parse_mode = parseMode.name
        return this
    }

    companion object {
        private const val serialVersionUID = 0L
    }
}
